### DISCUSSION TITLE
TypeError: only integer tensors of a single element can be converted to an index

### DISCUSSION THREAD

**Discussion by lakiet1609** [mrdbourke/pytorch-deep-learning#506] [Q&A]
Why after i load this code:    

```python
    model_path = r'models\model_2.pth'
    loaded_model_2 = cnn_model.custom_model(input_param=channels, output_param=num_classes) 
    # Load in the saved state_dict()
    loaded_model_2.load_state_dict(torch.load(f=model_path))
    # Send model to GPU
    loaded_model_2 = loaded_model_2.to(device)
```
Then i use pred_and_plot_image. 

It has the type error like the titile and it happens the error at this line of pred_and_plot_image
function:
```
plt.title(f"Pred: {class_names[target_image_pred_label]} | Prob: {target_image_pred_probs.max():.3f}")
```

**Comment by mrdbourke** [comment]
Hi @lakiet1609 ,

Did you check the value of `target_image_pred_label`?

It may be because it's not a valid index.

I'd step through your code to make sure the inputs and outputs are in the right data format.

E.g. it looks like `target_image_pred_label` should be an integer rather than a `torch.Tensor`.
